src/components/Dashboard/Tiles/Apps/AppTile.tsx

```diff
@@ -25,7 +26,35 @@ export const AppTile = ({ className, app }: AppTileProps) => {
     .filter((e) => e)
     .join(': ');
 
+  const parsedUrl = useMemo(() => {
+    try {
+      return tldts.parse(window.location.toString());
+    } catch {
+      return null;
+    }
+  }, [window.location]);
+
   const isRow = app.appearance.positionAppName.includes('row');
+  const externalUrl = useMemo(() => {
+    if (app.behaviour.externalUrl.length > 0) {
+      if (app.behaviour.externalUrl.startsWith('[homarr_base]')) {
+        const baseUrl = `${window.location.protocol}//${window.location.hostname}`;
+        return app.behaviour.externalUrl.replace('[homarr_base]', baseUrl);
+      } else if (
+        ['[homarr_hostname]', '[homarr_domain]', '[homarr_protocol]'].some((e) =>
+          app.behaviour.externalUrl.includes(e)
+        ) &&
+        parsedUrl
+      ) {
+        return app.behaviour.externalUrl
+          .replace('[homarr_hostname]', parsedUrl.hostname ?? '')
+          .replace('[homarr_domain]', parsedUrl.domain ?? '')
+          .replace('[homarr_protocol]', window.location.protocol.replace(':', ''));
+      }
+      return app.behaviour.externalUrl;
```

Comment:
The if statements seem a bit unnecessary, I would directly replace any mention to it's value and let the user make the mistake if need be. Adding a test without letting the user know might just cause confusion.
Also, you don't know what kind of crazy things people will want to try and do with their links after that.
It's fine as it is though, will still function as intended as long as the user uses it as intended too.
```suggestion
      return app.behaviour.externalUrl
        .replace('[homarr_base]', `${window.location.protocol}//${window.location.hostname}`)
        .replace('[homarr_hostname]', parsedUrl.hostname ?? '')
        .replace('[homarr_domain]', parsedUrl.domain ?? '')
        .replace('[homarr_protocol]', window.location.protocol.replace(':', ''));
```